parsers/etl_conventions.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python3
+
+# Parser for PEDSnet ETL conventions document
+
+import json
+import re
+import sys
+
+_field = re.compile(r'^(?P<name>[^\|]*)\|(?P<req>[^\|]*)'
+                    r'\|(?P<type>[^\|]*)\|(?P<desc>[^\|]*)'
+                    r'(\|(?P<conv>[^\|]*))?\|?$')
+_title = re.compile(r'^\#[^\#]+')
+_table_header = re.compile(r'^Field\s*\|\s*Required\s*\|')
+_table_name = re.compile(r'^\#\#\s*\d+\.\d+(?P<name>.*)$')
+
+
+def _get_table_name(l):
+    return _table_name.match(l).group('name').lower().strip()
+
+
+def _get_field(l):
+    '''
+    Read field information. The format is
+    'Field |Required | Data Type | Description | PEDSnet Conventions',
+    where the 'PEDSnet Conventions' field is optional.
+    '''
+    field = _field.match(l)
+
+    if not field:
+        return None
+
+    name = field.group('name').lower().strip()
+    req = field.group('req').lower().strip().startswith('yes')
+    data_type = field.group('type').strip()
+    desc = field.group('desc').strip()
+    conv = field.group('conv')
+    if conv:
+        conv = conv.strip()
+    else:
+        conv = ''
+
+    return (name, req, data_type, desc, conv)
+
+
+class Document():
+    def __init__(self, fileobj):
+        self.model = {
+            'content': '',
+            'tables': {}
+        }
+
+        self.fileobj = fileobj
+        self.line = ''
+
+        self.get_content()
+
+        while self.get_next_table():
+            pass
```

Comment:
These two methods are the entrypoint for the parser. Rather than calling them in the constructor and calling `get` later, let's combine them into a `parse` method:

```python
def parse(self):
    self.get_content()

    while self.get_next_table():
        pass

    return self.model
```

The caller can then initialize the document and call `parse` when ready.